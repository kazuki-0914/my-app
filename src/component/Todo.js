import React, { Component } from 'react';
export default class Todo extends Component {
    constructor(props) {
        super(props);
        this.state = {
            todos: [],
            name: '',
            // input: ''
        };
    }
    // onInput = (e) => {
    //     this.setState({
    //         name: e.target.value
    //     });
    // }
    addTodo = () => {
        const { todos, name } = this.state;
        this.setState({
            todos: [...todos, name]
        });
        this.setState({
            name: ''
        });
    }
    removeTodo = (index) => {
        const { todos, name } = this.state;
        this.setState({
            todos: [...todos.slice(0, index), ...todos.slice(index + 1)]
        });
    }
    updateValue = (e) => {
        // setInput(e.target.value);
        this.setState({
            name: e.target.value
        });
      };
    render() {
        const { todos } = this.state;

        return (<div>
            <input 
              data-testid="todo"
            type="text" 
            // onInput={this.onInput} 
            // placeholder="Enter" 
            value={this.state.name} 
            onChange={this.updateValue}
            />
            <button onClick={this.addTodo} data-testid="submit">登録</button>
            <ul>
                {todos.map((todo, index) =>
                    <li key={index}   >
                        タスク{index + 1}：
                        {todo}
                        <button data-testid="delete" onClick={() => { this.removeTodo(index) }}>削除</button>
                    </li>)}
            </ul>
        </div>);
    }
}
