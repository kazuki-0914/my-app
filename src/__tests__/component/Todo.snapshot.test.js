'use strict';
import { render, screen, fireEvent } from '@testing-library/react';
// import TestRenderer,{renderer } from 'react-test-renderer';
import Todo from '../../component/Todo';
import renderer from 'react-test-renderer';

const component = renderer.create(
  <Todo />
)

// beforeAll(() =>  render(<Todo />))
// beforeEach(() =>  {
//   const component = renderer.create(
//     <Todo />
//   )
// })
// afterEach(() => server.resetHandlers())
// afterAll(() => server.close())

// 初期表示
test('初期表示', async () => {
  // const component = renderer.create(
  //   <Todo />
  // )
  expect(component.toJSON()).toMatchSnapshot();
});

// 登録、削除
test('登録、削除', async () => {
  // const component = renderer.create(
  //   <Todo />
  // )
  // 登録
  component.getInstance().updateValue({ target: { value: 'Todo1' } });
  component.getInstance().addTodo();
  component.getInstance().addTodo();
  expect(component.toJSON()).toMatchSnapshot();
  // 削除
  component.getInstance().removeTodo(0);
  expect(component.toJSON()).toMatchSnapshot();
});
