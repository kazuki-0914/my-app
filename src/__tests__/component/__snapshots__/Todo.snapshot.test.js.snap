// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`初期表示 1`] = `
<div>
  <input
    data-testid="todo"
    onChange={[Function]}
    placeholder="Enter"
    type="text"
    value=""
  />
  <button
    data-testid="submit"
    onClick={[Function]}
  >
    登録
  </button>
  <ul />
</div>
`;

exports[`登録、削除 1`] = `
<div>
  <input
    data-testid="todo"
    onChange={[Function]}
    placeholder="Enter"
    type="text"
    value=""
  />
  <button
    data-testid="submit"
    onClick={[Function]}
  >
    登録
  </button>
  <ul>
    <li>
      タスク
      1
      ：
      Todo1
      <button
        data-testid="delete"
        onClick={[Function]}
      >
        削除
      </button>
    </li>
    <li>
      タスク
      2
      ：
      <button
        data-testid="delete"
        onClick={[Function]}
      >
        削除
      </button>
    </li>
  </ul>
</div>
`;

exports[`登録、削除 2`] = `
<div>
  <input
    data-testid="todo"
    onChange={[Function]}
    placeholder="Enter"
    type="text"
    value=""
  />
  <button
    data-testid="submit"
    onClick={[Function]}
  >
    登録
  </button>
  <ul>
    <li>
      タスク
      1
      ：
      <button
        data-testid="delete"
        onClick={[Function]}
      >
        削除
      </button>
    </li>
  </ul>
</div>
`;
